windows/code/RNDeviceInfoCPP.h

```diff
@@ -75,6 +153,79 @@ namespace winrt::RNDeviceInfoCPP
         });
     }
 
+    REACT_SYNC_METHOD(isKeyboardConnectedSync);
+    bool isKeyboardConnectedSync() noexcept
+    {
+      auto keyboardCapabilities = winrt::Windows::Devices::Input::KeyboardCapabilities();
+      return keyboardCapabilities.KeyboardPresent();
+    }
+
+    REACT_METHOD(isKeyboardConnected);
+    void isKeyboardConnected(ReactPromise<bool> promise) noexcept
+    {
+      promise.Resolve(isKeyboardConnectedSync());
+    }
+
+    REACT_SYNC_METHOD(isMouseConnectedSync);
+    bool isMouseConnectedSync() noexcept
+    {
+      auto mouseCapabilities = winrt::Windows::Devices::Input::MouseCapabilities();
+      return mouseCapabilities.MousePresent();
+    }
+
+    REACT_METHOD(isMouseConnected);
+    void isMouseConnected(ReactPromise<bool> promise) noexcept
+    {
+      promise.Resolve(isMouseConnectedSync());
+    }
+
+    // Helper function, should eventually be removed/replaced when it's exposed through RNW's public API.
+    bool IsXamlIsland() {
+      AppPolicyWindowingModel e;
+      // This value comes from an inline function GetCurrentThreadEffectiveToken() from processthreadsapi.h
+      auto token = (HANDLE)(LONG_PTR) -6;
+      auto windowingModel = CALL_INDIRECT(L"Api-ms-win-appmodel-runtime-l1-1-2.dll", AppPolicyGetWindowingModel, token, &e);
+      if (FAILED(windowingModel) || e == AppPolicyWindowingModel_ClassicDesktop) {
+        return true;
+      }
+      return false;
+    }
+
+    REACT_METHOD(isTabletMode);
+    void isTabletMode(ReactPromise<bool> promise) noexcept
+    {
+      if (IsXamlIsland()) {
+        // If the function succeeds, the return value is the requested system metric or configuration setting.
+        // If the function fails, the return value is 0. GetLastError does not provide extended error information.
+        auto ret = CALL_INDIRECT(L"ext-ms-win-ntuser-sysparams-ext-l1-1-0.dll", GetSystemMetrics, SM_TABLETPC);
```

Comment:
maybe we want SM_DIGITIZER here, I think SM_TABLETPC might always be false for win10 (I don't have a tablet to try this on at the moment)

Reply:
Updated!